### DISCUSSION TITLE
Usage for anonymization?

### DISCUSSION THREAD

**Discussion by dscarmo** [Emory-HITI/Niffler#105] [Q&A] [answered]
Hello! 

I was looking at this framework and wondering if there is a way to use it for anonymization of locally stored .dcm series?
Thank you

**Comment by pradeeban** [comment]
Hi,
You can convert *.dcm files to *.png files, with their respective metadata extracted as csv files (mapped to the png files, so that you know which png belongs to which original dcm) using https://github.com/Emory-HITI/Niffler/tree/master/modules/png-extraction

This way, you are effectively anonymizing by converting the *.dcm into *.png. Hope that helps your use case.

**Reply by dscarmo** [reply]
I always wondered if png conversion wouldn't reduce the image quality/information, if i remember correctly png requires conversion to the 0-255 uint8 range right? I work with medical image processing thats why that might be relevant.

**Reply by pradeeban** [reply]
True. You can run it and see whether the produced images are of acceptable data quality for you. For our use cases, it was proven to be sufficient for the past 2 years of medical image processing use cases, for our radiology faculty.

One major pitfall is when you specifically need an anonymized dicom file, rather than a png file (for example, if you want to feed the anonymized images back to your enterprise PACS). There are some enterprise anonymization frameworks (I don't remember the names. But you might be able to find from Google search) that do such anonymization while still maintaining them as dicom files. But Niffler is not capable of that, as of now.

We read the metadata and binary data. In theory, writing the metadata back after anonymizing PHI, into DICOM (rather than converting it into a PNG) shouldn't be hard. Just we never thought of it. But I will make it as an action item (if someone interested in contributing in the future).

**Reply by dscarmo** [reply]
For my use case the files still being .dcm would be ideal. It is hard to find a universal solution for that since there are so many different .dcm configurations/fields etc. Thanks for the quick answers!

**Reply by pradeeban** [reply]
I am tracking this discussion as https://github.com/Emory-HITI/Niffler/issues/106 now as a request for enhancement. I don't know when will this be implemented. But hopefully, someone will contribute to the implementation.

**Comment by pradeeban** [comment, marked as answer]
Fixed in Niffler-0.6.6, @dscarmo

https://github.com/Emory-HITI/Niffler/tree/dev/modules/dicom-anonymization

Please let us know if something is missing.
